ReactWindows/ReactNative.Shared/Common/AccessibilityHelper.cs

```diff
@@ -0,0 +1,229 @@
+        /// </summary>
+        /// <param name="element"></param>
+        /// <param name="value"></param>
+        public static void SetImportantForAccessibilityAttached(UIElement element, ImportantForAccessibility value)
+        {
+            element.SetValue(ImportantForAccessibilityAttachedProperty, value);
+        }
+
+        /// <summary>
+        /// ImportantForAccessibilityAttached property getter.
+        /// </summary>
+        /// <param name="element"></param>
+        /// <returns></returns>
+        public static ImportantForAccessibility GetImportantForAccessibilityAttached(UIElement element)
+        {
+            return (ImportantForAccessibility)element.GetValue(ImportantForAccessibilityAttachedProperty);
+        }
+
+        /// <summary>
+        /// Recursively sets AccessibilityView to accessibilityView for parentAutomationPeer children.
+        /// </summary>
+        /// <param name="parentAutomationPeer"></param>
+        /// <param name="accessibilityView"></param>
+        private static void SetChildrenAccessibilityView(AutomationPeer parentAutomationPeer, AccessibilityView accessibilityView)
+        {
+            if (parentAutomationPeer?.GetChildren() == null)
+            {
+                return;
+            }
+
+            foreach (AutomationPeer childAutomationPeer in parentAutomationPeer.GetChildren())
+            {
+                if (childAutomationPeer is FrameworkElementAutomationPeer childFEAutomationPeer)
+                {
+                    if (childFEAutomationPeer?.Owner != null)
+                    {
+                        AutomationProperties.SetAccessibilityView(childFEAutomationPeer.Owner, accessibilityView);
+                    }
+                }
+                else if (childAutomationPeer is ItemAutomationPeer childIAutomationPeer)
+                {
+                    if (childIAutomationPeer?.Item is UIElement childUIElement)
+                    {
+                        AutomationProperties.SetAccessibilityView(childUIElement, accessibilityView);
+                    }
+                }
+
+                SetChildrenAccessibilityView(childAutomationPeer, accessibilityView);
+            }
+        }
+
+        /// <summary>
+        /// Returns true if in uiElementAutomationPeer parents chain there is an UIElement with
+        /// ImportantForAccessibility value of Yes or NoHideDescendants.
+        /// </summary>
+        /// <param name="uiElementAutomationPeer"></param>
+        /// <returns></returns>
+        private static bool IsHiddenElement(AutomationPeer uiElementAutomationPeer)
+        {
+            if (uiElementAutomationPeer == null)
+            {
+                return false;
+            }
+
+            ImportantForAccessibility importantForAccessibilityAttached = ImportantForAccessibility.Auto;
+            var parentAutomationPeer = uiElementAutomationPeer.Navigate(AutomationNavigationDirection.Parent) as AutomationPeer;
```

Comment:
>var parentAutomationPeer = uiElementAutomationPeer.Navigate(AutomationNavigationDirection.Parent) as AutomationPeer [](start = 12, length = 115)

maybe it would be clearer if we do this AutomationPeer as another "else if" so that we have all cases as is branches?

also, maybe do the "null" as the final "else" in the "if"?

also maybe we can use new switch pattern matching if our language level in the project allows it and if we'll actually move it from Shared into ReactNative so we don't need to support WPF variant?

Reply:
I reworked the code to use switch with pattern matching and split into all possible cases including AutomationPeer, null and default, so that all cases are listed. Should make it easier to maintain.

---
In reply to: [163112637](https://github.com/Microsoft/react-native-windows/pull/1595#discussion_r163112637) [](ancestors = 163112637)